linkatos/parser.py

```diff
@@ -7,48 +7,73 @@ def is_empty(message_list):
     return ((message_list is None) or (len(message_list) == 0))
 
 
+def capture_reaction(sub_message):
+    parsed = {
+        'message': sub_message['reaction'],
+        'channel': sub_message['item']['channel'],
+        'item_ts': sub_message['item']['ts'],
+        'type': 'reaction',
+        'user': sub_message['user'],
+        'item_user': sub_message['item_user']
+    }
+    return parsed
+
+
+def capture_url(sub_message, url):
+    parsed = {
+        'message': url,
+        'channel': sub_message['channel'],
+        'ts': sub_message['ts'],
+        'type': 'url',
+        'user': sub_message['user']
+    }
+    return parsed
+
+
 def parse(input_message, BOT_ID):
     """
         The Slack Real Time Messaging API is an events firehose.
         this parsing function returns None unless:
         1. someone posts a url starting with httpS?//
            in this case it returns the url and a out_type = 'url'
-        2. a yes or a no after a url
-           in this case it returns True for a yes, False for a no
-        The function also extracts the name of the channel
+        2. someone adds a thumbsup or a thumbsdown
+           in this case it returns the reaction and the type of outcome
     """
-    print(input_message)  # print the list of outputs to get them on screen
+    print('input_message:', input_message)  # print the list of outputs to get them on screen
 
     # default outcome
-    parsed = {'out': None, 'channel': None, 'type': None}
+    parsed = {
+        'message': None,
+        'channel': None,
+        'ts': None,
+        'item_ts': None,
+        'type': None,
+        'user': None,
+        'item_user': None
+    }
 
+    # if the message list is empty return an empty object
     if is_empty(input_message):
-        return (parsed)  # empty output
+        return (parsed)
 
     for sub_message in input_message:
```

Comment:
Initially it was because I saw the structure of the messages were part of an array of `sub_messages`. However I'm not sure how to produce a multi sub-message message.
I'll change it to message.